ratis-test/src/test/java/org/apache/ratis/datastream/TestDataStreamBase.java

```diff
@@ -65,8 +84,7 @@ public int write(ByteBuffer src) {
         }
         final int remaining = src.remaining();
         for(; src.remaining() > 0; ) {
-          Assert.assertEquals(pos2byte(byteWritten), src.get());
-          byteWritten += 1;
+          Assert.assertEquals(pos2byte(byteWritten.getAndIncrement()), src.get());
```

Comment:
Sorry, why we use AtomicInteger for byteWritten, will `write(ByteBuffer src)` be called by multi-thread ?

Reply:
Using int may work but it is hard to understand why it works.  Since this is a test, we want to make sure that it won't has bugs.  When the test fails, we don't want to ask questions like "do we need to synchronize byteWritten?"  :)

Reply:
@szetszwo  `public int write(ByteBuffer src) ` seems won't be called by multi-thread in NettyServerStreamRpc currently ?

Reply:
In a async world (Netty), all methods are run by some threadpools.  You are right that write(ByteBuffer src) will be synced externally.  I will change it back to int.

Reply:
We can change it to AtomicInteger when a state machine is somehow shared by multiple stream.  Current tests are not set up in this way.

Though I guess a state machine shared by multiple stream is the real world pattern, e.g. multiple clients stream concurrently to a Raft group.

Reply:
@amaliujia  Each stream has its own byteWritten.